### DISCUSSION TITLE
How to disable lazy loading for components used with Nuxt Content?

### DISCUSSION THREAD

**Discussion by joulev** [nuxt/content#2174] [Q&A]
I notice that in one of my page using Nuxt Content, the components used in the Markdown renderer are only lazy loaded when they are needed. For example in the following page, I hide certain elements and only show them, using `<ContentRenderer :value="slide" />` when I click some buttons (\*). As you see, the components are only loaded when they are needed.

<details>
<summary>Screen recording</summary>

https://github.com/nuxt/content/assets/44609036/9f4e0cbf-1bf4-42bb-bdc1-9b1b1ca78c02

(`CustomComponent` is a component I defined in `components/content`)
</details>

Normally this is a very good feature, but for this one time I would like to load *all* Nuxt Content components on page load – I would like to sacrifice initial page load so that components are available instantly when I need them. How can I do that?

I'm actually not even sure if this is related to Nuxt Content, as I'm quite new to Nuxt. If you need more info do let me know – I can't provide a minimal reproduction repository at the moment but will try to if it is necessary.

---

(\*) The showing/hiding mechanism is like this

```vue
<!-- wrapper.vue -->
<div v-if="isActive">
  <slot />
</div>
```

```vue
<!-- main.vue -->
<Wrapper
  v-for="(content, i) in props.contents"
  :key="i"
  :is-active="activeIndex === i"
>
  <ContentRenderer :value="content" />
</Wrapper>
```

I did not explicitly opt into any lazy loading; and the way I see it, although the component is hidden in the tree, I still run `ContentRenderer` for the entire content of the page (not just the visible part), so I would like the required components to be loaded on page load as well.

**Comment by nobkd** [comment]
PR #2221 is probably related, but I don't think, that all the components will be loaded on initial page load with only this...
